docs/src/userguide/numpy_tutorial.rst

```diff
@@ -417,31 +432,55 @@ know what NumPy data type we should use for our output array.
 
 In this case, our function now works for ints, doubles and floats.
 
-.. literalinclude:: ../../examples/userguide/numpy_tutorial/convolve_fused_types.pyx
-    :linenos:
+.. literalinclude:: ../../examples/userguide/numpy_tutorial/compute_fused_types.pyx
 
 We can check that the output type is the right one::
 
-    >>>naive_convolve_fused_types(f, g).dtype
+    >>>compute(array_1, array_2, a, b, c).dtype
     dtype('int32')
-    >>>naive_convolve_fused_types(f.astype(np.double), g.astype(np.double)).dtype
+    >>>compute(array_1.astype(np.double), array_2.astype(np.double), a, b, c).dtype
     dtype('float64')
 
 We now do a speed test:
 
 .. sourcecode:: ipython
 
-    In [25]: %timeit convolve_fused_types.naive_convolve(f, g)
-    20 ms ± 392 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
+    In [25]: %timeit compute_fused_types.compute(array_1, array_2, a, b, c)
+    6.17 ms ± 164 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
+
+We're a bit slower than before, because of the right call to the clip function
+must be found at runtime and adds a bit of overhead.
+
+Using multiple threads
+======================
+
+Cython have support for OpenMP. It have also some nice wrappers around it,
+like the function :func:`prange`. You can see more information about Cython and
+parralelism in :ref:`parallel`. Since we do elementwise operations, we can easily
+distribute the work among multiple threads. It's important not to forget to pass the
+correct arguments to the compiler to enable OpenMP. When using the Jupyter notebook,
+you should use the cell magic like this::
+
+    %%cython --compile-args=-fopenmp --link-args=-fopenmp --force
+
+The GIL must be released (see :ref:`Releasing the GIL <nogil>`), so this is why we
+declare our :func:`clip` function ``nogil``.
+
+.. literalinclude:: ../../examples/userguide/numpy_tutorial/compute_prange.pyx
+
+We can have substantial speed gains for minimal effort:
+
+.. sourcecode:: ipython
+
+    In [25]: %timeit compute_prange.compute(array_1, array_2, a, b, c)
+    3.55 ms ± 80.6 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
 
-We're still around 800 times faster than the interpreted Python version.
+We're now 7858 times faster than the pure Python version and 2.3 times faster
+than NumPy!
```

Comment:
This would probably improve further with larger arrays and/or more CPU cores. 4ms are already so short that the thread startup time and all that becomes visible overhead. I understand that it's not easy to have this example scale up because the Python code runtimes are already half a minute, but maybe 5x the array size would give more impressive numbers?